components/tor/tor_launcher_factory.cc

```diff
@@ -75,7 +75,9 @@ void TorLauncherFactory::Init() {
       &TorLauncherFactory::OnTorCrashed, weak_ptr_factory_.GetWeakPtr()));
 }
 
-TorLauncherFactory::~TorLauncherFactory() {}
+TorLauncherFactory::~TorLauncherFactory() {
+  std::move(*control_.release()).DeleteSoon();
```

Comment:
this is pretty confusing notation. Why can't we just call the method by pointer?
So instead of just properly calling `DeleteSoon`, we properly call it *and* add mind-blowing boilerplate, that basically doesn't provide any guarantees

Reply:
because calling it by pointer doesn't ensure that it is released by the owner and it does provide guarantees because trying to call it any other way will result in a compilation failure. Also not calling it at all will result in a DCHECK.

Reply:
addressed in https://github.com/brave/brave-core/pull/7927/commits/536325b0aca2613cdb3ca3e1c5deac2a25f7dfd9